### PR TITLE
Drop record viewer refactor

### PR DESCRIPTION
**Pull request by myangelkamikaze**
- [x] refactor `DropRecordRow` - this currently supports both regular rows and merged rows, which makes it very complicated and hard to follow
- [x] apply common drop filter to merged results
- [x] remove old searcher code
- [ ] future idea: let users make their own filters for which ships to include in the results

### CODE REVIEW

File: ElectronicObserver/Data/Constants.cs [hunk 1]
```diff
@@ -1001,20 +991,12 @@ public static string GetExpeditionResult(int value)
 	#endregion
 
 	#region Servers
-	public class KCServer
+	public class KCServer(int num, string name, string jp, string ip)
```
[the remark this answers is not in the document]
Comment (myangelkamikaze) [reply ElectronicObserver/Data/Constants.cs:994]: Probably, but that one is actually unrelated to drop record viewer, I only made the change cause sonar was complaining because I updated editorconfig.
So I wouldn't do that refactoring in this PR.

[the remark this answers is not in the document]
Comment (Jebzou) [reply ElectronicObserver/Data/Constants.cs:994]: okay

File: ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs [hunk 2]
```diff
@@ -3,9 +3,12 @@
 using System.Collections.ObjectModel;
 using System.ComponentModel;
```
Comment (Jebzou) [inline ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs:4]: unused using

File: ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs [hunk 3]
```diff
@@ -3,9 +3,12 @@
...
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 using System.IO;
 using System.Linq;
 using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
 using System.Windows.Controls;
 using System.Windows.Forms;
```
Comment (Jebzou) [inline ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs:13]: unused using

File: ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs [hunk 4]
```diff
@@ -15,31 +18,32 @@
 using ElectronicObserver.Common.Datagrid;
 using ElectronicObserver.Data;
 using ElectronicObserver.Resource.Record;
+using ElectronicObserver.Utility;
 using ElectronicObserver.ViewModels;
 using ElectronicObserver.ViewModels.Translations;
+using ElectronicObserver.Window.Dialog.QuestTrackerManager.Enums;
 using ElectronicObserver.Window.Dialog.ShipPicker;
 using ElectronicObserver.Window.Wpf;
```
Comment (Jebzou) [inline ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs:26]: unused using

File: ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs [hunk 5]
```diff
@@ -349,341 +360,292 @@ private string GetMapString(int serialID, bool insertEnemyFleetName = true)
...
-					continue;
+		if (MapDifficulty is int difficulty && difficulty != r.Difficulty) return false;
 
+		return true;
+	}
 
-				if (MapAreaID is int world && world != r.MapAreaID) continue;
-				if (MapInfoID is int map && map != r.MapInfoID) continue;
-				if (MapCellID.Ids?.Contains(r.CellID) == false) continue;
+	[SuppressMessage("Critical Code Smell", "S3776:Cognitive Complexity of methods should not be too high", Justification = "<Pending>")]
```
Comment (Jebzou) [inline ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs:430]: i think you can remove the cognitive complexity warning with return early

Comment (myangelkamikaze) [reply ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs:430]: nope
image

File: ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs (hunk 4 above)
Comment (myangelkamikaze) [reply ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs]: fixed

File: ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs [hunk 6]
```diff
@@ -3,9 +3,12 @@
...
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 using System.IO;
 using System.Linq;
 using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
 using System.Windows.Controls;
 using System.Windows.Forms;
 using System.Windows.Media;
```
[the remark this answers is not in the document]
Comment (myangelkamikaze) [reply ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs]: fixed

File: ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs (hunk 3 above)
Comment (myangelkamikaze) [reply ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs]: fixed

File: ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs (hunk 2 above)
Comment (myangelkamikaze) [reply ElectronicObserver/Window/Tools/DropRecordViewer/DropRecordViewerViewModel.cs]: fixed

**Review by Jebzou — APPROVED** [review]
